tests/functional/aws-node-sdk/test/quota/tooling.js

```diff
@@ -0,0 +1,48 @@
+const fetch = require('node-fetch');
+const AWS = require('aws-sdk');
+const xml2js = require('xml2js');
+
+const sendRequest = async (method, host, path, body = '', callback) => {
+    const service = 's3';
+    const endpoint = new AWS.Endpoint(host);
+
+    const request = new AWS.HttpRequest(endpoint);
+    request.method = method.toUpperCase();
+    request.path = path;
+    request.body = body;
+    request.headers.Host = host;
+    request.headers['X-Amz-Date'] = new Date().toISOString().replace(/[:\-]|\.\d{3}/g, '');
+    const sha256hash = AWS.util.crypto.sha256(request.body || '', 'hex');
+    request.headers['X-Amz-Content-SHA256'] = sha256hash;
+    request.region = 'us-east-1';
+
+    const signer = new AWS.Signers.V4(request, service);
+    signer.addAuthorization(AWS.config.credentials, new Date());
+
+    const url = `http://${host}${path}`;
+    const options = {
+        method: request.method,
+        headers: request.headers
+    };
+
+    if (method !== 'GET' && method !== 'HEAD') {
+        options.body = request.body;
+    }
+
+    try {
+        const response = await fetch(url, options);
+        const text = await response.text();
+        const result = await xml2js.parseStringPromise(text);
+        if (result && result.Error) {
+            callback(result, null);
+        } else {
+            callback(null, result);
+        }
```

Comment:
```suggestion
        if (result && result.Error) {
            return callback(result, null);
        }
        return callback(null, result);
```

Also it's strange to have just one object passed in both places, can't we have an error and a result? (or return a promise)

Reply:
done here : https://github.com/scality/cloudserver/pull/5552/commits/d2eafe4aa67d06489918c52c68e02a26fcb01446#diff-fb6cdfa9407a28bd369de18ea7f6948d2f3f04b5b9975dddc68bc809b8825529R33-R45